Content.Server/Chat/V2/Systems/ChatRateLimitSystem.cs

```diff
@@ -0,0 +1,105 @@
+﻿using Content.Server.Administration.Logs;
+using Content.Server.Chat.Managers;
+using Content.Shared.CCVar;
+using Content.Shared.Chat.V2;
+using Content.Shared.Chat.V2.Systems;
+using Content.Shared.Database;
+using Content.Shared.Players;
+using Robust.Shared.Configuration;
+using Robust.Server.Player;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Chat.V2.Systems;
+
+public sealed class ChatRateLimitSystem : EntitySystem
```

Comment:
For @ShadowCommander , @Chief-Engineer et al - I'm doing validation and rate limiting is a part of it. Please eyeball this. You might want to lift and shift some of this code before PCgamertide?